src/datasources/asset-calibration/AssetCalibrationDataSource.ts

```diff
@@ -151,6 +166,19 @@ export class AssetCalibrationDataSource extends DataSourceBase<AssetCalibrationQ
     }
   }
 
+  private async loadSystems(): Promise<void> {
+    if (this.state.systems) {
+      return;
+    }
+
+    try {
+      const systems = await this.querySystems('', ['id', 'alias','connected.data.state', 'workspace']);
+      this.state.systems = new Map(systems.map(system => [system.id, system]))
+    } catch (error) {
+      throw new Error(`An error occurred while querying systems: ${error}`);
```

Comment:
Here we are losing the call stack because we are re-throwing the error with a different message that doesn't give anything valuable. I would just let the exception bubble up to have easier tracing.